### ISSUE TITLE
Keep having trouble with HTTP header... failed request

### ISSUE THREAD

**Issue by KMZoilus** [Huntereb/Awoo-Installer#47]

**Comment by Huntereb** [comment]
Do you use Mac OS?

**Comment by KMZoilus** [comment]
nope , windows 10 64B

**Comment by madh77** [comment]
> keep trying to connect but get but get this in the fail window:
> 
> **PerformRequest93: Failed to retrieve HTTP header. could not connect to server.**
> 
> My pc is wired to router, switch is on wifi... any ideas? The usb install when connected to pc is great! but wireless is a problem.

I have your same problem but I've find a solution.
So this method worked for me cause i'm using a mac with a windows 10 x64 on virtual machine via VMware fusion.
So if this is your same condition this method can worked for you too.
This problem is related to the ip address connection on mac, so if you use windows on VM, go to VM settings > network adapter > change the selected box from "share with my mac" to the box "wifi". So now the windows VM create a dedicated connection with no issue. Now the installation via Network with Awoo Installer should work out. Hope this solution can worked for you and the developer.

**Comment by dreamfalcon** [comment]
Same here with win10 64bits.
After selecting to install on the SD card, it changes to installing but after a few moments gives this error.

**Comment by Couchy** [comment]
Same here using NS-USBload v2.2. It sends the lists the list of nsps fine, but fails trying to install them. I've disabled the firewall on Windows and my router to no avail.

**Comment by Ali-Asim800** [comment]
Yeah i've got this issue too and it needs to be solved.  I was really looking fordward to installing my games because for some reason my switch isn't detected on my computer and because of that i can't do wirless either. this issue needs to be fixed please. I'm using ns-usbloader and a windows computer. Maybe i should try and connect my pc to ethernet.

**Comment by Huntereb** [comment]
This is likely a firewall issue pc-side. Try disabling Windows firewall, or whatever Mac OS uses for a firewall.

**Comment by Ali-Asim800** [comment]
Ok I will try that Huntereb, thank you for your help. I will reach out to you again if it still doesn't work.


Kind Regards
 Ali Asim
________________________________

**Comment by Ali-Asim800** [comment]
it worked thanks for reaching out to help.

**Comment by soyeliel** [comment]
> This is likely a firewall issue pc-side. Try disabling Windows firewall, or whatever Mac OS uses for a firewall.

Yeap, same problem on MAC and disable my antivirus did the job.

**Comment by JoshuaManuel** [comment]
Not sure if this is related, but I've been having an issue installing from both Google Drive and URLs. It shouldn't matter that I'm on MacOS because the switch shouldn't even be connecting to it. I'm sure the files are publicly accessible. I get the following error:

HTTPDownload:157: Failed to retrieve HTTP Header: SSL connect error.

I've tried going into the code to see if I can get more info on this with easy_curl's errorbuffers, but no luck.

**Comment by donjonne** [comment]
> Not sure if this is related, but I've been having an issue installing from both Google Drive and URLs. It shouldn't matter that I'm on MacOS because the switch shouldn't even be connecting to it. I'm sure the files are publicly accessible. I get the following error:
> 
> HTTPDownload:157: Failed to retrieve HTTP Header: SSL connect error.
> 
> I've tried going into the code to see if I can get more info on this with easy_curl's errorbuffers, but no luck.

same here, did you find any solution?  i believe the first two versions worked.  is it possible to rollback or will sig patches not allow it?

**Comment by Huntereb** [comment]
@JoshuaManuel @donjonne this is because you're using a Switch with its prodinfo wiped. This will be fixed on the next release.

**Comment by donjonne** [comment]
> @JoshuaManuel @donjonne this is because you're using a Switch with its prodinfo wiped. This will be fixed on the next release. (You will have to manually update)

huntereb
i actually got it running a while back ago
sorry for not giving updates
here is something i found:
1.  for some reason my computer pc gave me the letter "I" (big i) the same as "l" (small L)
so finding this out it caught the correct link
2.  i believe i was not connected to the internet one time(what else could it have been idk)
-is there a way to incorporate the wifi signal in the app as to probe that user is indeed with internet connection
3.  quota for original g drive link was exceeded
-just made a copy to my personal drive and made a public link


i believe the last problem is the most common one tbh

**Comment by Huntereb** [comment]
@donjonne You can't access the net install page if you don't have access to a wifi connection. You may have disconnected while entering the ID or something. Just try again next time. :P

**Comment by donjonne** [comment]
> @donjonne You can't access the net install page if you don't have access to a wifi connection. You may have disconnected while entering the ID or something. Just try again next time. :P

yea very odd
anyways all is working good since
last game i installed was a xcz thru a public gdrive link for the game: "awesome italian 3d guy" lol

**Comment by gammesB** [comment]
for me the problem in the end was that windows was blocking ns-usbloader. had to cancel the blockage in properties
![image](https://user-images.githubusercontent.com/96754537/162594069-77396059-67c0-4171-8e44-9f8f5d1462a1.png)

**Comment by dannyglover** [comment]
> for me the problem in the end was that windows was blocking ns-usbloader. had to cancel the blockage in properties
> ![image](https://user-images.githubusercontent.com/96754537/162594069-77396059-67c0-4171-8e44-9f8f5d1462a1.png)
> 

This was my issue too. Thank you for sharing the solution.

**Comment by damu-u** [comment]
I had the same issue on my m1 mac os and my problem was due to the VPN being turned on. I turned off the active VPN and it worked fine.

**Comment by aforty** [comment]
Anyone else coming across this in 2023, I was able to fix this by disabling "Windows Defender Firewall" on my private network.

**Comment by Jimmytheprofessionaldumbass** [comment]
> > This is likely a firewall issue pc-side. Try disabling Windows firewall, or whatever Mac OS uses for a firewall.
> 
> Yeap, same problem on MAC and disable my antivirus did the job.


how did you disable the antivirus? is built into macos? also idk if this helps but I somehow wasn't even connected to the same wifi, but by putting in the ip listed on awoo, it was able to send the game file name over correctly, but when sending the file I get the listed issue. even though it wasn't connected on the same wifi, my mac was connected to my hotspot, which was connected to my home router, and my switch was connected to my home router as well, maybe the hotspot redirects or something else is going on? but yeah, its weird. Hope this info helps anyhow
